services/src/main/java/org/keycloak/protocol/oidc/endpoints/TokenEndpoint.java

```diff
@@ -503,10 +504,17 @@ public Response resourceOwnerPasswordCredentialsGrant() {
                 .setSession(session)
                 .setUriInfo(uriInfo)
                 .setRequest(request);
-        Response challenge = processor.authenticateOnly();
-        if (challenge != null) {
-            cors.build(httpResponse);
-            return challenge;
+        try {
+            Response challenge = processor.authenticateOnly();
+
+            if (challenge != null) {
+                cors.build(httpResponse);
+                return challenge;
+            }
+        } catch (AuthenticationFlowException e) {
+        	// Can occur if the wrong protocol is used, user is under brute force protection, etc.
+            event.error(Errors.IDENTITY_PROVIDER_ERROR);
+            throw new CorsErrorResponseException(cors, OAuthErrorException.ACCESS_DENIED, "Access denied", Response.Status.UNAUTHORIZED);
```

Comment:
This is masking the real problem and just using a generic access denied error which doesn't seem correct either

Reply:
Sure.  So this wasn't there before, and the token Endpoint today just throws a 500 if an AuthenticationFlowException occurs during the user auth process.  I can just put the behavior back to be what it is today...